enginetest/enginetests.go

```diff
@@ -4996,6 +4996,80 @@ func TestAddDropPks(t *testing.T, harness Harness) {
 			{1, 1},
 		}, nil, nil)
 	})
+
+	if _, ok := harness.(*MemoryHarness); ok {
+		t.Skip("in memory tables don't implement sql.Rewritable yet")
+	}
+
+	t.Run("Add primary key column with auto increment", func(t *testing.T) {
+		ctx.SetCurrentDatabase("mydb")
+		RunQuery(t, e, harness, "CREATE TABLE t2 (i int, j int);")
+		RunQuery(t, e, harness, "insert into t2 values (1,1), (2,2), (3,3)")
+		AssertErr(
+			t, e, harness,
+			"alter table t2 add column pk int primary key;",
+			sql.ErrPrimaryKeyViolation,
+		)
+
+		TestQueryWithContext(
+			t, ctx, e, harness,
+			"alter table t2 add column pk int primary key auto_increment;",
+			[]sql.Row{{sql.NewOkResult(0)}},
+			nil, nil,
+		)
+
+		TestQueryWithContext(
+			t, ctx, e, harness,
+			"select pk from t2;",
+			[]sql.Row{
+				{1},
+				{2},
+				{3},
+			},
+			nil, nil,
+		)
+
+		TestQueryWithContext(
+			t, ctx, e, harness,
+			"show create table t2;",
+			[]sql.Row{
+				{"t2", "CREATE TABLE `t2` (\n  `i` int,\n  `j` int,\n  `pk` int NOT NULL AUTO_INCREMENT,\n  PRIMARY KEY (`pk`)\n) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_0900_bin"},
+			},
+			nil, nil,
+		)
+	})
+
+	t.Run("Add primary key column with auto increment first", func(t *testing.T) {
```

Comment:
Have a test with a float too

Reply:
according to this warning, mysql plans on deprecating auto_increment float, do you still want me to add a test?
```sql
MySQL > alter table t add column pk float primary key auto_increment;
Query OK, 0 rows affected, 1 warning (0.0405 sec)

Records: 0  Duplicates: 0  Warnings: 1
Warning (code 3856): AUTO_INCREMENT support for FLOAT/DOUBLE columns is deprecated and will be removed in a future release. Consider removing AUTO_INCREMENT from column 'pk'.
```

Reply:
We support it in dolt (and in GMS) so we should have a test for it